ecs.tf

```
@@ -5,12 +5,13 @@ data "aws_ecs_cluster" "target_cluster" {
 resource "aws_ecs_task_definition" "main_task" {
   family                   = "${var.app_name}-tsk"
   requires_compatibilities = [var.launch_type]
-  network_mode             = "bridge"
+  network_mode             = var.launch_type == "FARGATE " ? "awsvpc" : var.network_mode
```

If the launch type isn't required to be all caps for fargate then you might want to consider something like this:
```suggestion
  network_mode             = lower(var.launch_type) == "fargate " ? "awsvpc" : var.network_mode
```

---

Though if it has to be all caps this is fine already ofc.

---

Pretty sure it's required to be all caps. All the docs reference it as such, although admittedly I haven't tried it in lower.

---

Alternately, I guess you could use `upper` just to be safe.

---

Is it possible to only allow specific values for inputs to modules? You could limit the launch type to be exactly `FARGATE` or `EC2` if you can, which removes this as a problem.

---

https://stackoverflow.com/questions/57190035/terraform-how-to-restrict-an-input-variable-to-a-list-of-possible-choices
Looks like we can, and give a custom error message too which is pretty neat.

---

Looks like the closet you can get is [Custom Validation Rules](https://developer.hashicorp.com/terraform/language/values/variables)

---

Added validation for launch type